tools/e2e-commons/bin/wait-for-atomic-site.cjs

```diff
@@ -0,0 +1,114 @@
+const path = require( 'path' );
+const fetch = require( 'node-fetch' );
+
+// Below call should be BEFORE requiring config, so library wil pick it up.
+process.env.NODE_CONFIG_DIR = path.resolve( __dirname, '../config' );
+const { resolveSiteUrl, getSiteCredentials } = require( '../helpers/utils-helper.cjs' );
+
+function getAuthHeader() {
+	const credentials = getSiteCredentials();
+	return (
+		'Basic ' +
+		Buffer.from( credentials.username + ':' + credentials.apiPassword ).toString( 'base64' )
+	);
+}
+
+async function getJetpackVersionFromSite() {
+	let response;
+	try {
+		response = await fetch( resolveSiteUrl() + '/index.php?rest_route=/wp/v2/plugins', {
+			headers: { Authorization: getAuthHeader() },
+		} );
+
+		const contentType = response.headers.get( 'content-type' );
+		if ( contentType && ! contentType.startsWith( 'application/json' ) ) {
+			const out = await response.text();
+			if ( out.includes( 'Briefly unavailable for scheduled maintenance' ) ) {
+				console.log( 'Site is down for maintenance' );
+			}
+			return {};
+		}
+		const plugins = await response.json();
+
+		const jetpackDev = plugins.find( p => p.plugin === 'jetpack-dev/jetpack' );
+		// console.log(jetpackDev);
+		return jetpackDev.version;
+	} catch ( error ) {
+		console.error( `Logging environment details failed! ${ error }` );
+		return '';
+	}
+}
+
+async function forcePluginUpdates() {
+	const response = await fetch(
+		resolveSiteUrl() + '/index.php?rest_route=/jp-e2e/v1/beta-autoupdate',
+		{
+			method: 'POST',
+			headers: { Authorization: getAuthHeader() },
+		}
+	);
+	console.log( await response.json() );
+}
+
+async function getLatestVersion() {
+	const type = getVersionType();
+	const response = await fetch( 'https://betadownload.jetpack.me/jetpack-branches.json' );
+	const manifest = await response.json();
+
+	if ( type === 'rc' || type === 'trunk' || type === 'master' ) {
+		return manifest[ type ].version;
+	}
+	return manifest.pr.type.version;
+}
+
+async function waitForPluginUpdate( expectedVersion ) {
+	let timesRun = 0;
+	const interval = setInterval( async () => {
+		console.log( 'Checking for update' );
+		const jpVersion = await getJetpackVersionFromSite();
+		if ( expectedVersion === jpVersion ) {
+			console.log( 'Update completed' );
+			process.exit( 0 );
+		}
+		timesRun += 1;
+
+		if ( timesRun > 20 ) {
+			console.error( 'Was running for ' + timesRun * 20 + ' seconds, exiting' );
+			clearInterval( interval );
+			process.exit( 1 );
+		}
+	}, 5000 );
+}
+
+function getVersionType() {
+	const refType = process.argv[ 2 ];
+	const refName = process.argv[ 3 ];
+
+	if ( refName === 'trunk' && refType === 'branch' ) {
```

Comment:
We'll also have events for release branches, I'm not sure if you intentionally omitted those.

Reply:
I think the release approach is changing a bit, for example, now we have a `prerelease` branch: https://github.com/Automattic/jetpack-production/actions/runs/3533205023/jobs/5928533826

I'm not sure how to handle these cases. Meaning is there a new beta artifact created? Similarly, for release branches, I'm not sure that every commit to the release branch actually creates a beta artifact as well. I'd prefer to limit the scope of this PR on trunk & tags events, if that's ok.

Reply:
I think it's ok to limit to trunk and tags, but the problem is that using any other branch as argument results in `Invalid version type` and fails the build. I think we should probably skip the atomic tests in this case. 

GitHub won't let me suggest a change in an untouched file, but here's where you could add a ref type and ref name check:
 https://github.com/Automattic/jetpack/blob/51c396f47ce3853b1b47c44d76c027261c3d1f5c/.github/files/e2e-tests/e2e-matrix.js#L92

Reply:
Oh right! I think I'll update the script's exit code and leave a TODO to add support for other branches in the future.